packages/components/src/autocomplete/index.js

```diff
@@ -418,53 +418,47 @@ function useAutocomplete( {
 	};
 }
 
+function useLastDifferentValue( value ) {
+	const history = useRef( new Set() );
+
+	history.current.add( value );
```

Comment:
This is a bit weird, in React 18, rerendering can happen even without any valid reason (suspended component that reruns). So I believe these history related changes should be done in an effect somehow?

Reply:
Why is that a problem? If the value is the same reference, it's a no-op?

Reply:
I don't understand what this hook is supposed to do, but the thing is that you'll keep adding values to this history where you might not need it?

Reply:
What if we use `usePrevious` instead of this hook? Is this the same thing?

Reply:
The hook uses a `Set`, so if it re-renders, it will be with the same value and the hook will return the same result. So to me it looks fine.

> I don't understand what this hook is supposed to do
> What if we use usePrevious instead of this hook? Is this the same thing?

This hook will return the last value that was different. `usePrevious` just returns the last value even if it's the same one.

Reply:
Tried to understand this better, I think it's broken. So imagine you have this:

 - First render "value = 1" (hook returns 1) correct
 - Second render "value = 2" (hook returns 1) correct
 - For some reason React suspended the component from rendering and retriggered the rendering with the same value.
 - Third render "value = 2" (hooks returns 2) but it should return 1.
 
 Am I wrong?

Reply:
I think it's wrong? Why would it return 2? The history still has `Set( 1, 2)`. Another render with the same value doesn't change that. The hook will still return 1.

Reply:
This is why I'm using a Set. We're not just pushing a value to an array on every render. If there's a re-render with the same value, the history doesn't change at all.

Reply:
For me, the set will contain `( 1, 2, 2 )` after the add, then the delete removes the first element `1` so it becomes `(2, 2)` and then we'll return `2`. It's possible that I'm reading this wrong.

Reply:
Ok, I'm wrong here, the `Set` only keeps unique values. Ignore me :)